cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -501,8 +501,10 @@ func ValidateNetworking(c *kubeadm.ClusterConfiguration, fldPath *field.Path) fi
 	}
 	if len(c.Networking.PodSubnet) != 0 {
 		allErrs = append(allErrs, ValidateIPNetFromString(c.Networking.PodSubnet, constants.MinimumAddressesInPodSubnet, isDualStack, field.NewPath("podSubnet"))...)
-		// Pod subnet was already validated, we need to validate now against the node-mask
-		allErrs = append(allErrs, ValidatePodSubnetNodeMask(c.Networking.PodSubnet, c, field.NewPath("podSubnet"))...)
+		if c.ControllerManager.ExtraArgs["allocate-node-cidrs"] != "false" {
```

Comment:
we need unit tests cmd/kubeadm/app/phases/controlplane/manifests_test.go

Reply:
Hi @aojea !
I'm sorry for the delay.
Could you please have a look?

Reply:
If `allocation-node-cidrs` is not explicitly configured, it will be defaulted as `true` by kubeadm.
So we should validate the Pod subnet against the node-mask when `c.ControllerManager.ExtraArgs["allocate-node-cidrs"] != "false"`.

Reply:
I think is correct but let's double check with @neolit123 ^^^

Reply:
the  KCM flag `allocate-node-cidrs` is `false` by default, so it makes sense to validate in kubeadm only if the the user has set the flag to `true`.

Reply:
If the `PodSubnet` is set and the `ExtraArgs["allocate-node-cidrs"]` is not set, kubeadm will add `allocate-node-cidrs=true` flag for KCM automatically.
In this case,  can we skip this validation? 
Or should we not add the `allocate-node-cidrs` flag for KCM automatically?
https://github.com/kubernetes/kubernetes/blob/43e8405d9c675521eff6e19c9bedbe116eb43d9e/cmd/kubeadm/app/phases/controlplane/manifests.go#L307-L308

Reply:
the KCM docs say that `cluster-cidr` requires `allocate-node-cidrs`, so we should be setting it to `true` in the default args. adding the flag by default seems correct to me. this make it easier for the user and not require them to pass flags when they wish to set a pod subnet.

extraArgs are merged over default args.
so if the user was explicit about `allocate-node-cidrs==false`, we should respect it and skip validation in ValidateNetworking.

so the diff here seems valid to me.